apps/src/lib/node/ledger/shell/prepare_proposal/block_space_alloc/states.rs

```diff
@@ -0,0 +1,173 @@
+//! All the states of the [`BlockSpaceAllocator`] state machine,
+//! over the extent of a Tendermint consensus round
+//! block proposal.
+//!
+//! # States
+//!
+//! The state machine moves through the following state tree:
+//!
+//! 1. [`BuildingDecryptedTxBatch`] - the initial state. In
+//!    this state, we populate a block with DKG decrypted txs.
+//! 2. [`BuildingProtocolTxBatch`] - the second state. In
+//!    this state, we populate a block with protocol txs.
+//! 3. [`BuildingEncryptedTxBatch`] - the third state. In
+//!    this state, we populate a block with DKG encrypted txs.
+//!    This state supports two modes of operation, which you can
+//!    think of as two states diverging from [`BuildingProtocolTxBatch`]:
+//!   * [`WithoutEncryptedTxs`] - When this mode is active, no encrypted txs are
+//!     included in a block proposal.
```

Comment:
When would we expect to use this `WithoutEncryptedTxs` mode? Since currently all regular txs given to us by Tendermint are "encrypted"